runtime/domain.c

```diff
@@ -778,6 +778,7 @@ static void domain_create(uintnat initial_minor_heap_wsize,
 reallocate_minor_heap_failure:
   caml_teardown_major_gc();
 init_major_gc_failure:
+  caml_orphan_shared_heap(d->state->shared_heap);
```

Comment:
Minor: I suspect this call is actually a no-op because there should be no major-heap allocation during domain initialization (after heap initialization and before the domain starts running mutator code), but it does not hurt to have it for robustness (maybe later the domain-initialization code will start using the major heap?).